### PR TITLE
300mm-x-extruder

### PR DESCRIPTION
**Pull request by SGr33n**
### Which mods are added by this PR?

300mm X Extruder endcaps

### How was it tested?

Printing it and using on my own Enderwire

### Any background context you want to provide?

I was not able to chop the X Extruder so I decided to buy a 300mm version, which is smaller than the one suggested on the main guide, so I designed that endcaps to fill the gaps.

### ⚠️⚠️**PLEASE READ AND COMPLETE THE CHECKLIST BELOW**⚠️⚠️

  * [ yes ] I have read the rules available [here](https://github.com/VoronDesign/VoronUsers/wiki/Mod-Submission-Rules) and
        my mod adheres to these rules.
  * [ yes ] This mod was created by myself and I agree to publish it under the repository
        [license](https://github.com/VoronDesign/VoronUsers/blob/master/LICENSE.md)

### CODE REVIEW

File: printer_mods/sgr33n/300mm-x-extruder-endcap/.metadata.yml [hunk 1]
```diff
@@ -0,0 +1,31 @@
...
+# A short description of your mod. Try to describe your mod with 1-2 short sentences
+description: This mod makes stabler and prettier the use of a 300mm profile for the X-Extruder.
+# Mod version: Set this to 1 when you are submitting a new mod. When you make changes to your mod, you may increment this number
+mod_version: 1
+# You may add your discord username here so people can contact you about your mod
+discord_username: sgr33n
+# The CAD files for your mod. Please provide them as a list.
+# Note that each mod must include at least one CAD file.
+cad:
+- CAD/300mm_X-Extruder_Endcap.step
```
Comment (mjonuschat) [inline printer_mods/sgr33n/300mm-x-extruder-endcap/.metadata.yml:13]: These two files are reasonable large, if you could try compressing them as a ZIP to see if we can get them down to below 2MB each that would be great.

File: printer_mods/sgr33n/300mm-x-extruder-endcap/STL/x_extruder_endcap_left.stl
Comment (mjonuschat) [inline printer_mods/sgr33n/300mm-x-extruder-endcap/STL/x_extruder_endcap_left.stl:1]: This file does not look like it's printable without supports due to the embossed logo

**Review by mjonuschat — CHANGES_REQUESTED** [review]
Thanks for submitting your mod. There are a few minor issues in the metadata.yml file that should be easy to correct.
The bigger issue is the printability of the left endcap with the embossed logo, it doesn't look like it's easily printable without supports, could you take a look at that and either add some instructions on how to correctly print it without supports to the readme or make some changes that improve printability.

More a question than a request for change: Would it make sense to rename the mod to "300m X-Extrusion Endcap" to help discoverability?

File: printer_mods/sgr33n/300mm-x-extruder-endcap/STL/x_extruder_endcap_left.stl
Comment (SGr33n) [reply printer_mods/sgr33n/300mm-x-extruder-endcap/STL/x_extruder_endcap_left.stl:1]: Thanks for your reply @mjonuschat It's 0.9mm inset, should print without support. Alternatively which measure would be good for you to not need supports? I could also remove the logo at all.

Comment (mjonuschat) [reply printer_mods/sgr33n/300mm-x-extruder-endcap/STL/x_extruder_endcap_left.stl:1]: That is weird, both the tooling here and my slicer show it as a raised logo on the left endcap:

Screenshot 2024-05-16 at 07 25 23

Comment (SGr33n) [reply printer_mods/sgr33n/300mm-x-extruder-endcap/STL/x_extruder_endcap_left.stl:1]: Sorry, you're right, my fault!
Fixed.

File: printer_mods/sgr33n/300mm-x-extruder-endcap/.metadata.yml [hunk 2]
```diff
@@ -0,0 +1,30 @@
...
+# The parts of your mod, exported and oriented ready to be sliced. Please provide them as a list.
+# Note that each mod must include at least one STL or OBJ file.
+stl:
+- STL/x_extruder_endcap_left.stl
+- STL/x_extruder_endcap_right.stl
+# The image files provided with your mod. These may be instructions you use in your README.md file, or other additional pictures.
+# **Note that the first image specified here will be used as cover image on mods.vorondesign.com**
+# All paths are relative to this .metadata.yml file.
+images:
+- Images/300mm-X-Extruder-Endcap-X-Carriage.jpg
```
Comment (mjonuschat) [inline printer_mods/sgr33n/300mm-x-extruder-endcap/.metadata.yml:23]: It looks like `gallery-1.jpg` and `gallery-2.jpg` are missing from this list.

File: printer_mods/sgr33n/300mm-x-extruder-endcap/STL/x_extruder_endcap_left.stl
Comment (mjonuschat) [reply printer_mods/sgr33n/300mm-x-extruder-endcap/STL/x_extruder_endcap_left.stl:1]: Thanks for fixing this. Looks like we only have the two images missing from the metadata left to resolve

File: printer_mods/sgr33n/300mm-x-extruder-endcap/.metadata.yml (hunk 2 above)
Comment (SGr33n) [reply printer_mods/sgr33n/300mm-x-extruder-endcap/.metadata.yml:23]: Added to .metafile.yml, thanks :)

File: printer_mods/sgr33n/300mm-x-extruder-endcap/.metadata.yml (hunk 1 above)
Comment (SGr33n) [reply printer_mods/sgr33n/300mm-x-extruder-endcap/.metadata.yml:13]: I've cleaned the step file of unnecessary components.

**Review by mjonuschat — APPROVED** [review]
Thanks for making those changes, looks good to me.

**Comment by SGr33n** [comment]
Good, thanks :)

**Review by chirpy2605 — APPROVED** [review]
LGTM

**Comment by thetic** [comment]
Extruder -> Extrusion? Current naming is very confusing.

**Comment by SGr33n** [comment]
> Extruder -> Extrusion? Current naming is very confusing.

Extrusion, you're right.
I'll make the changes, can I also modify the dir name?

**Comment by SGr33n** [comment]
> > Extrusion, you're right. I'll make the changes, can I also modify the dir name?
> 
> Yes, rename the dir also, so that everything matches, and then 👍 from me!

I also renamed the branch name, shall I open a new pull request?
Thanks

**Comment by MapleLeafMakers** [comment]
> > > Extrusion, you're right. I'll make the changes, can I also modify the dir name?
> > 
> > 
> > Yes, rename the dir also, so that everything matches, and then 👍 from me!
> 
> I also renamed the branch name, shall I open a new pull request? Thanks

No need, this is good, thanks for your submission!

**Pull request merged**